### ISSUE TITLE
[feature request] memchr_inv?

### ISSUE THREAD

**Issue by cyphar** [BurntSushi/memchr#166]
Sometimes it's necessary to search for the first occurrence that is *not* a particular byte (the most obvious example is checking if a buffer is all NUL-bytes and returning the first non-NUL byte). For instance, [the Linux kernel has a `memchr_inv` function for this purpose](https://elixir.bootlin.com/linux/v6.11.3/source/lib/string.c#L820).

The usecase I have at the moment is that I'm writing a C API for [libpathrs](https://github.com/openSUSE/libpahtrs) that I would like to design to mirror a Linux kernel API, and `memchr_inv` is needed to check whether a passed buffer is zeroed after a point. I have hand-rolled `memchr_inv` but I suspect it is far less efficient than something that this crate could provide.

(I guess `memchr` already handles this, but it's probably worth mentioning that the buffer is probably not going to be 4-byte or 8-byte aligned because offset from where we have to check the buffer could be any value and so the slice could start at any offset.)

**Comment by BurntSushi** [comment]
Yeah, I think what I wrote in #157 still applies. I do _want_ to add things like this (and stuff in #157). And one way of avoiding the compile time hit is to make them opt-in features, particularly since they are somewhat niche needs. I think the bigger problem is review and maintenance bandwidth.

**Comment by larseggert** [comment]
+1 that something like `memchr_inv` would be great for efficiently skipping over sequences of zeros (which in my use case - QUIC - are padding frames in a packet).
